### PR TITLE
fix: Simplify / update build instructions.

### PR DESCRIPTION
**Pull request by lizroth**
*Issue #, if available:*

*Description of changes:* Update build instructions. Reference the AWS SDK for C++ build instructions where possible to deduplicate setup. Simplify the steps for the `aws-c-common` branch.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Review by alex-chew — APPROVED** [review]

File: README.md
Comment (juneb) [inline README.md:12]: Can we be very clear here? "Building the AWS Encryption SDK for C (with support for AWS KMS)"

(The current statement can be interpreted as using KMS to help with the build.)

File: README.md
Comment (juneb) [inline README.md:15]: Let's define "fresh". Also s/installs/installations. 

I love the voice, really, but many of our customers are global. Let's be clear for them.

File: README.md
Comment (juneb) [inline README.md:20]:
```suggestion
The Windows instructions install everything in your current directory (where you run the build process). To change the installation directory, see the Tips and Tricks section at the end of this README.
```

File: README.md
Comment (juneb) [inline README.md:23]: s/KMS/AWS KMS throughout, please.

File: README.md
Comment (juneb) [inline README.md:36]:
```suggestion
For best results, create an EC2 instance with an instance profile that at includes AWS KMS permissions for Encrypt, Decrypt, and GenerateDataKey on
at least one KMS CMK in your account. You will not need any other AWS
```

File: README.md
Comment (juneb) [inline README.md:49]:
```suggestion
The AWS SDK for C++ and the AWS Encryption SDK for C share dependencies on OpenSSL, aws-c-common, and CMake. The AWS SDK for C++ has additional dependencies and prerequisites. See [AWS SDK for
C++: Getting Started](https://github.com/aws/aws-sdk-cpp#getting-started).
```

File: README.md
Comment (juneb) [inline README.md:51]: Delete "both." There are more than two things.

File: README.md
Comment (juneb) [inline README.md:55]:
```suggestion
To build shared libraries, specify the `-DBUILD_SHARED_LIBS=ON` flag to build
```

File: README.md
Comment (juneb) [inline README.md:60]:
```suggestion
First, build the AWS SDK for C++. That will install the shared dependencies.
```

File: README.md
Comment (juneb) [inline README.md:63]:
```suggestion
If you need the AWS SDK for C++ only to use the AWS Encryption SDK, you can build only the AWS KMS SDK.
To do so, add the `-DBUILD_ONLY="kms"` flag to `cmake` as noted in the [AWS SDK for C++ build instructions](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/setup-linux.html).
```

File: README.md
Comment (juneb) [inline README.md:65]: Delete. Moved this into the previous sentence.

File: README.md
Comment (juneb) [inline README.md:76]: s/obtain/get. Also s/on/for (nit).

File: README.md
Comment (juneb) [inline README.md:85]:
```suggestion
need the AWS SDK for C++ only to use the AWS Encryption SDK, you can build only the AWS KMS SDK.
See [these directions](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/setup-linux.html#setup-linux-from-source)
and specify the `-DBUILD_ONLY="kms"` flag to `cmake` in the instructions provided.
```

File: README.md
Comment (juneb) [inline README.md:96]: What docker-images folder? Link? Is there any reason not to use the automation? Should we mark the alternate steps as manual?

File: README.md
Comment (juneb) [inline README.md:98]: s/Start by installing/Install

File: README.md
Comment (juneb) [inline README.md:100]: Why use that command prompt? Is it the only one that works? Or the only one we tested?

File: README.md
Comment (juneb) [inline README.md:111]: Does the specified command do the integration? If so:

To integrate vcpkg and Visual Studio (optional), use the following command: `.\vcpkg\vcpkg integrate install`

File: README.md
Comment (juneb) [inline README.md:113]:
```suggestion
Build the AWS SDK for C++. This installs the aws-c-common dependency, too. If you need the AWS SDK for C++ only to use the 
```

Place "only" as close to the reference as possible.

File: README.md
Comment (juneb) [inline README.md:114]:
```suggestion
AWS Encryption SDK, you can build only the AWS KMS SDK. To build the entire AWS SDK for C++, delete `-DBUILD_ONLY=kms` from ???.
```

To do A, do B. (for left-to-right languages)

File: README.md
Comment (juneb) [inline README.md:128]: Building the AWS Encryption SDK for C (without support for AWS KMS)

File: README.md
Comment (juneb) [inline README.md:133]: s/KMS/AWS KMS (per kenbeer@)

File: README.md
Comment (lizroth) [reply README.md:108]: This is one of the things that I was unhappy with and am fixing, so hold this thought please for the next round

File: README.md
Comment (lizroth) [reply README.md:63]: I put the instructions in this order with the intent that I state the required step (first, build the SDK); then I explain what modifications the customer must make to the instructions ("if you only need..."); then I finally link to the directions after clarifying what needs to change.

My concern about putting the instructions link first/combined with the previous sentence is that the customer will correctly click the link and go follow those instructions without realizing that our instructions have a couple of notes that apply to those steps.

Does that make sense? Should I still change this, or change it in some different way?

File: README.md
Comment (lizroth) [reply README.md:94]: https://github.com/aws/aws-encryption-sdk-c/tree/master/docker-images

Requiring customers to use Docker in particular creates another dependency to get started working with the SDK. These generic instructions guide customers through bootstrapping their project generically on any system.

I'll add a link

File: README.md
Comment (lizroth) [reply README.md:98]: It's the one to have permissions to install the libraries in the default path and to have all of the developer tools populated on the command line without needing to modify or specify the path

**Review by lizroth — COMMENTED** [review]
Most feedback has been incorporated -- a couple of follow up comments/questions to your feedback so far. Thank you!

**Comment by texastony** [comment]
According to Alex Chew, this PR is awaiting approval from June due to Doc Impact. 

We have an open issue (#717) that is related to this.

Also, we will have to manually rebase this.

**Comment by texastony** [comment]
Errr... #127 may not be related... it appears the requester has successfully built the library, but cannot use the library in any other project... which is interesting...

**Comment by feliperodri** [comment]
cc. @alex-chew

File: README.md
Comment (juneb) [reply README.md:63]: That's fine.

File: README.md
Comment (juneb) [reply README.md:98]: Makes sense.

**Pull request merged**
